android/app/src/main/java/com/ecomap/ecomap/MainActivity.kt

```diff
@@ -24,21 +54,202 @@ class MainActivity : AppCompatActivity() {
             insets
         }
 
-        store = UserStore(this.applicationContext)
-
         // User token validation.
-        val intent = Intent(this, SignInActivity::class.java)
+        val intentSignInActivity = Intent(this, SignInActivity::class.java)
 
         // Flags the intent to mark the activity as the root in the history stack,
         // clearing out any other tasks.
-        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
+        intentSignInActivity.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
 
+        // Check whether the user store contains the login token.
+        // If not, start the SignIn Activity.
+        val store = UserStore(this.applicationContext)
         runBlocking {
             val token = store.getToken().first()
             if (token == null) {
-                // Open SignIn Activity if token is not found.
-                startActivity(intent)
+                startActivity(intentSignInActivity)
+            }
+        }
+
+        // Construct the main entry point for the Android location services.
+        fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(this)
+
+        // Google Map configurations.
+        val mapLatLngBounds = LatLngBounds(
+            LatLng(MAP_CAMERA_BOUND_SW_LAT, MAP_CAMERA_BOUND_SW_LNG),
+            LatLng(MAP_CAMERA_BOUND_NE_LAT, MAP_CAMERA_BOUND_NE_LNG)
+        )
+
+        val googleMapOptions = GoogleMapOptions()
+        googleMapOptions
+            .mapType(GoogleMap.MAP_TYPE_NORMAL)
+            .latLngBoundsForCameraTarget(mapLatLngBounds)
+
+        // Add support map fragment to the map container.
+        val mapFragment = SupportMapFragment.newInstance(googleMapOptions)
+        supportFragmentManager
+            .beginTransaction()
+            .add(R.id.fragment_container_view_map, mapFragment)
+            .commit()
+
+        // Register the map callback.
+        mapFragment.getMapAsync(this)
+
+        // Get activity views.
+        val buttonMyLocation: FloatingActionButton = findViewById(R.id.button_my_location)
+
+        // Set button functions.
+        buttonMyLocation.setOnClickListener { focusMyLocation() }
+    }
+
+    /**
+     * Function called when the Google Map is ready.
+     */
+    override fun onMapReady(googleMap: GoogleMap) {
+        this.map = googleMap
```

Comment:
```suggestion
        map = googleMap
```
For consistency purposes, the keyword `this` can be removed.